### PR TITLE
Add more detail to CONTRIBUTING

### PR DESCRIPTION
**Pull request by kw217**
Pick up details from #38 and record for posterity.

### CODE REVIEW

**Comment by kw217** [comment]
Sorry it looks like it all changed - I spotted it had mostly DOS line endings! Fixed to Unix now.

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -1,78 +1,94 @@
...
+code which exhibits the issue.
+
+## Code Contribution
+If you wish to contribute code to this project, you are welcome to do so
+using a github pull request, however you will need to sign each commit,
+see details below.
+
+A few important notes:
+
+* Please use `cargo fmt` to format the code. If it's not already properly formatted, please put the reformatting into a separate commit so it is easier to review; don't combine it with actual changes.
```
Comment (angusi) [inline CONTRIBUTING.md]: Nit: The rest of this document is wrapped at ~80 chars. Should this be wrapped too?

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -1,78 +1,94 @@
...
+* There are no tests in this repo, but there are examples which at least use most features. If it makes sense, when you're adding a feature please add to the examples.
+* Please do ensure you carry your changes through to https://github.com/Metaswitch/cassandra-rs and add tests there (if that makes sense).
+* In general: I know this code isn't as nice as it could be, but please ensure new code does things better, and if you have an opportunity to clean up old code then please do so.
+
+If updating the Cassandra driver version, there are examples at #18 and #39. The steps are:
+  * Update `cassandra.h` from the latest driver.
+  * Regenerate the binding code with bindgen utility, using `--size_t-is-usize` for compatibility.
```
Comment (angusi) [inline CONTRIBUTING.md]: IMO it would be helpful, especially for first-time contributors, to link through to the bindgen docs here.

**Review by angusi — APPROVED** [review]
LGTM, just two minor comments.

**Pull request merged**